cms/server/admin/handlers/user.py

```diff
@@ -278,3 +285,81 @@ def post(self, user_id):
+            generate_passwords = self.get_body_argument('generate_passwords',
+                                                        False)
+            ignored = 0
+            try:
+                user_csv = self.request.files["users_csv"][0]
+                users = CsvUserLoader(None, None, user_csv['body']) \
+                    .get_users(generate_passwords=generate_passwords)
+                processed_users = []
+                for user in users:
+                    db_user = self.sql_session.query(User).filter_by(
+                        username=user.username).first()
+                    if db_user:
+                        if ignore_existing:
+                            if db_user.password:
+                                db_user.password = 'not_empty'
+                            processed_users.append((False, db_user))
+                            ignored += 1
+                        else:
+                            self.application.service.add_notification(
+                                make_datetime(), 'Import failed',
+                                'User "%s" already exists' % user.username)
+                            self.redirect(fallback_page)
+                            return
+                    else:
+                        processed_users.append((True, user))
+
+                if ignored:
+                    self.application.service.add_notification(
+                        make_datetime(),
+                        "User exist:", '%d users already exist' % ignored)
+                r_params['users'] = processed_users
+                self.render("users_import_preview.html", **r_params)
+                return
+            except Exception as error:
+                self.application.service.add_notification(
+                    make_datetime(), "Bad csv file", repr(error))
+                self.redirect(fallback_page)
+                return
+        elif action == 'save':
+            usernames = self.get_body_arguments('username', None)
+            first_names = self.get_body_arguments('first_name', None)
+            last_names = self.get_body_arguments('last_name', None)
+            passwords = self.get_body_arguments('password', None)
+            for i in range(len(usernames)):
+                args = {
+                    'username': usernames[i],
+                    'first_name': first_names[i],
+                    'last_name': last_names[i],
+                }
+                if passwords[i]:
+                    args['password'] = passwords[i]
+                else:
+                    args['password'] = \
+                        cmscommon.crypto.generate_random_password()
+                user = User(**args)
+                self.sql_session.add(user)
+            if self.try_commit():
+                # Create the user on RWS.
+                self.application.service.proxy_service.reinitialize()
+                self.redirect("/users")
+                return
```

Comment:
Removed one of the else branches.  I'd like to keep a return after each redirect or page render as not much can be done afterwards. That way it is clearer at that point that everything is done and makes it impossible to accidentally render two pages results in an error.